appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -27,27 +28,45 @@ import com.appcues.util.ResultOf.Failure
 import com.appcues.util.ResultOf.Success
 import kotlinx.coroutines.flow.SharedFlow
 import org.koin.core.component.KoinScopeComponent
+import org.koin.core.component.get
 import org.koin.core.component.inject
 import org.koin.core.scope.Scope
 
 internal class ExperienceRenderer(
     override val scope: Scope,
-) : KoinScopeComponent {
+) : KoinScopeComponent, StateMachineOwning {
```

Comment:
do we need to have ExperienceRenderer implementing StateMachineOwning? or could wehave a separate Implementation inited ? for instance AppcuesModalStateMachineOwning

Reply:
I was trying to actually not create any new types, but just have `ExperienceRenderer` and `AppcuesFrameView` implement `StateMachineOwning` (similar to iOS side). This is not possible in Kotlin though, as a public type can use an internal Interface, but the properties that it adds must be public - and we cannot have a public StateMachine or RenderContext as those are internal types. This is why the `AppcuesFrameStateMachineOwning` got created - to have a separate internal type for that case. 

I think it would work fine to have a separate `AppcuesModalStateMachineOwning` as well, though I don't think it would actually get used for anything, and it also works to just have `ExperienceRenderer` implement the interface as done here - either way.